microsoft/testsuites/stress/stress_ng_suite.py

```diff
@@ -0,0 +1,161 @@
+
+
+@TestSuiteMetadata(
+    area="stress",
+    category="stress-ng",
+    description="""
+    A suite for running the various classes of stressors provided
+    by stress-ng.
+    """,
+)
+class StressNgTestSuite(TestSuite):
+    TIME_OUT = 3600
+    CONFIG_VARIABLE = "stress_ng_jobs"
+
+    @TestCaseMetadata(
+        description="""
+        Runs a stress-ng jobfile. The path to the jobfile must be specified using a
+        runbook variable named "stress_ng_jobs". For more info about jobfiles refer:
+        https://manpages.ubuntu.com/manpages/jammy/man1/stress-ng.1.html
+        """,
+        priority=4,
+    )
+    def perf_stress_ng_jobfile(
+        self,
+        variables: Dict[str, Any],
+        environment: Environment,
+        result: TestResult,
+    ) -> None:
+        if self.CONFIG_VARIABLE in variables:
+            jobs = variables[self.CONFIG_VARIABLE]
+            for job_file in jobs:
+                self._run_stress_ng_job(job_file, environment, result)
+        else:
+            raise SkippedException("No jobfile provided for stress-ng")
+
+    @TestCaseMetadata(
+        description="Runs stress-ng's 'cpu' class stressors for 60s each.",
+        priority=4,
+    )
+    def perf_stress_ng_cpu_stressors(
+        self,
+        environment: Environment,
+    ) -> None:
+        self._run_stressor_class(environment, "cpu")
+
+    @TestCaseMetadata(
+        description="Runs stress-ng's 'memory' class stressors for 60s each.",
+        priority=4,
+    )
+    def perf_stress_ng_memory_stressors(
+        self,
+        environment: Environment,
+    ) -> None:
+        self._run_stressor_class(environment, "memory")
+
+    @TestCaseMetadata(
+        description="Runs stress-ng's 'vm' class stressors for 60s each.",
+        priority=4,
+    )
+    def perf_stress_ng_vm_stressors(
+        self,
+        environment: Environment,
+    ) -> None:
+        self._run_stressor_class(environment, "vm")
+
+    @TestCaseMetadata(
+        description="Runs stress-ng's 'io' class stressors for 60s each.",
+        priority=4,
+    )
+    def perf_stress_ng_io_stressors(
+        self,
+        environment: Environment,
+    ) -> None:
+        self._run_stressor_class(environment, "io")
+
+    @TestCaseMetadata(
+        description="Runs stress-ng's 'network' class stressors for 60s each.",
+        priority=4,
+    )
+    def perf_stress_ng_network_stressors(
+        self,
+        environment: Environment,
+    ) -> None:
+        self._run_stressor_class(environment, "network")
+
+    def _run_stressor_class(self, environment: Environment, class_name: str) -> None:
+        nodes = [cast(RemoteNode, node) for node in environment.nodes.list()]
+        procs: List[Process] = []
+        for node in nodes:
+            procs.append(node.tools[StressNg].launch_class_async(class_name))
+        for proc in procs:
+            proc.wait_result(timeout=self.TIME_OUT, expected_exit_code=0)
```

Comment:
should we store the perf results if they are perf testing, or any other check, I want to figure out the value of these cases.

Reply:
Well, now that I think about it, they are not really perf tests. We don't collect any metrics. According to the stress-ng man page: 

> ... it has never been intended to be used as a precise benchmark test suite, so do NOT use it in this manner.

The value of these tests right now is to induce a high load in the VM and see if anything bad (such as kernel panic) happens.

In the future, we plan to add detection of other kinds of stack traces in the kernel logs as well. The current `Dmesg` tool has the capability to detect such things, however we need more fine-grained detection (e.g. "Call Trace" matches all traces but in this case, we want to ignore OOMs since they are expected while stressing).

Reply:
i would call they are stressor then perf. stress-ng is valuable tool for stress testing, usually used for long term testing.

Reply:
@anirudhrb we can pull the serial console log once VM is unresponsive in after_case.

Reply:
How will we know the VM is unresponsive? Also, I believe the libvirt platform captures the serial logs of the nodes for each lisa run. Is this not the case for Azure platform?